### PR TITLE
[SOUND] mmebuddy_debug.h: Sort out macro inclusions

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

`NDEBUG` should not ignore errors and assertions, as suggested by Oleg.

Follow-up of c7a4984 (0.4.15-dev-8414).

## Proposed changes

- POPUP() is used on DBG only.
- SND_ERR() and SND_ASSERT() are wanted even when defined(NDEBUG).
- Add a FIXME as !defined(NDEBUG) code compilation remains broken.
Note: That is a pre-existing issue, which I am not going to investigate.
- Match WCHAR type.

### CODE REVIEW

File: sdk/include/reactos/libs/sound/mmebuddy_debug.h [hunk 1]
```diff
@@ -4,29 +4,21 @@
...
-    #define SND_WARN(...) \
+    // Helper for SND_ASSERT().
+    #define POPUP(...) \
         { \
-            WCHAR dbg_popup_msg[1024]; \
+            WCHAR dbg_popup_msg[1024], dbg_popup_title[256]; \
+            wsprintf(dbg_popup_title, L"%hS(%d)", __FILE__, __LINE__); \
             wsprintf(dbg_popup_msg, __VA_ARGS__); \
-            OutputDebugString(dbg_popup_msg); \
+            MessageBox(0, dbg_popup_msg, dbg_popup_title, MB_OK | MB_TASKMODAL); \
```
Comment (HBelusca) [inline sdk/include/reactos/libs/sound/mmebuddy_debug.h:18]:
```suggestion
            MessageBoxW(NULL, dbg_popup_msg, dbg_popup_title, MB_OK | MB_TASKMODAL); \
```

File: sdk/include/reactos/libs/sound/mmebuddy_debug.h [hunk 2]
```diff
@@ -43,6 +35,29 @@
...
+
+#endif // DBG
+
+#if DBG && !defined(NDEBUG)
+
+    #define SND_WARN(...) \
+        { \
+            WCHAR dbg_popup_msg[1024]; \
+            wsprintf(dbg_popup_msg, __VA_ARGS__); \
+            OutputDebugString(dbg_popup_msg); \
```
Comment (HBelusca) [inline sdk/include/reactos/libs/sound/mmebuddy_debug.h:51]:
```suggestion
            OutputDebugStringW(dbg_popup_msg); \
```

File: sdk/include/reactos/libs/sound/mmebuddy_debug.h [hunk 3]
```diff
@@ -43,6 +35,29 @@
...
+        }
+
+    #define SND_TRACE(...) \
+        { \
+            WCHAR dbg_popup_msg[1024]; \
+            wsprintf(dbg_popup_msg, __VA_ARGS__); \
+            OutputDebugString(dbg_popup_msg); \
```
Comment (HBelusca) [inline sdk/include/reactos/libs/sound/mmebuddy_debug.h:58]:
```suggestion
            OutputDebugStringW(dbg_popup_msg); \
```

File: sdk/include/reactos/libs/sound/mmebuddy_debug.h [hunk 4]
```diff
@@ -4,29 +4,21 @@
...
             wsprintf(dbg_popup_msg, __VA_ARGS__); \
-            OutputDebugString(dbg_popup_msg); \
+            MessageBox(0, dbg_popup_msg, dbg_popup_title, MB_OK | MB_TASKMODAL); \
         }
-    #define SND_TRACE(...) \
+
+    #define SND_ERR(...) \
         { \
             WCHAR dbg_popup_msg[1024]; \
             wsprintf(dbg_popup_msg, __VA_ARGS__); \
```
Comment (HBelusca) [inline sdk/include/reactos/libs/sound/mmebuddy_debug.h:24]: below: `OutputDebugString` -> `OutputDebugStringW`

**Review by HBelusca — APPROVED** [review]

**Review by oleg-dubinskiy — APPROVED** [review]

**Pull request merged**